client/core/trade.go

```diff
@@ -1116,6 +1119,9 @@ func (t *trackedTrade) matchIsRevoked(match *matchTracker) bool {
 // If redeemMustConfirm is true, a match that has completed but the redemption
 // has not yet been confirmed will still be considered active.
 func (t *trackedTrade) matchIsActive(match *matchTracker, redeemMustConfirm bool) bool {
+	if match.needsDynamicSwapFeeConfirms || match.needsDynamicRedemptionFeeConfirms {
+		return true
+	}
```

Comment:
There is a TODO related to this https://github.com/decred/dcrdex/blob/e04a5d3ef44253041f2a1f81a64caf4dbdfa1bbf/client/core/trade.go#L1290-L1292

Reply:
If the swap is complete and we've confirmed the redeem as required by `redeemMustConfirm` (babysit ETH redemptions), then the order can be retired.

The swap and redeem fees would only be used when viewing the order in the order details page (or exporting order history I suppose).  The way that is done for inactive orders is by loading the order from DB just for that purpose (no longer in `dc.trades`).

So, after on-demand loading the inactive order and it's matches, can we check something as a heuristic to suggest if we need to recompute real fees instead of trusting `OrderMetaData.{SwapFeesPaid,RedemptionFeesPaid}`?  Even if that means adding a flag or something to the DB, at least that means we don't have to adjust our definition of order/match activeness.

Reply:
Well, I think I realized that as long as the redeem confirmed count is greater or equal to the number for updating the fees, we can remove this `match.needsDynamicSwapFeeConfirms || match.needsDynamicRedemptionFeeConfirms` condition and the DB reload issue is already solved.

Reply:
> DB reload issue is already solved.

Currently needs isSwappable or isRedeemable to return with ready true. If loading from db after the status is changed I think the funds will never be added still. But, other than that two confirm window I guess this issue is solved.

Reply:
Currently, isSwappable or isRedeemable need to return with ready ok once before we start looking for for the fee transaction. I think there is a two confirm window after that where if the client was shut down we would loose the check flags and the balance not updated if we restored from db because the swap state has already moved on.

Reply:
Saving to db... but not changing the confirm logic. So, it`ll find it if it finds it while active for other reasons. Should be fine.